### PR TITLE
Split liveness in two items

### PR DESCRIPTION
**Pull request by danielepolencic**

### CODE REVIEW

File: application-development.md [hunk 1]
```diff
@@ -19,25 +19,43 @@ The kubelet executes the check and decides if the container should be restarted.
...
-You can find [more details about using (or not) liveness probes in this article](https://blog.colinbreck.com/kubernetes-liveness-and-readiness-probes-revisited-how-to-avoid-shooting-yourself-in-the-other-foot/#letitcrash).
+Instead, you should immediately exit the process and let the kubelet restart the container.
+
+### Configure a passive Liveness probe
+
+The Liveness probe is designed to restart your container when it's stuck.
+
+Consider the following scenario: if your application is processing an infinite loop, there's no way to exit or ask for help.
+
+When the process is consuming 100% CPU, it won't have time to reply to the Readiness probe checks, and it will be eventually removed from the Service.
```
Comment (salmaniqbal) [inline application-development.md]: Should it say "it won't have time to reply to the liveness probe checks,...?

File: application-development.md [hunk 2]
```diff
@@ -19,25 +19,43 @@ The kubelet executes the check and decides if the container should be restarted.
...
+
+However, the Pod is actively counted against the active replicas for the current Deployment.
```
Comment (salmaniqbal) [inline application-development.md]: Maybe reword this slightly:
"However, the Pod is still registered as an active replicas for the current Deployment."

File: application-development.md [hunk 3]
```diff
@@ -19,25 +19,43 @@ The kubelet executes the check and decides if the container should be restarted.
...
+
+In other words, not only the process isn't serving requests, but also it consumes resources.
```
Comment (salmaniqbal) [inline application-development.md]: "In other words, not only the process isn't serving requests, it is also consuming resources."

File: application-development.md [hunk 4]
```diff
@@ -19,25 +19,43 @@ The kubelet executes the check and decides if the container should be restarted.
...
+
+You should expose and use a Liveness probe for the current container that replies with a success response.
+
+The probe should not have any logic, and it should be used as a recovery mechanism in case the process is not responsive.
```
Comment (salmaniqbal) [inline application-development.md]: "The probe should not have any logic. It should be used as a recovery mechanism in case the process is not responsive."

File: application-development.md [hunk 5]
```diff
@@ -19,25 +19,43 @@ The kubelet executes the check and decides if the container should be restarted.
...
+
+Consider the following scenario: if your application is processing an infinite loop, there's no way to exit or ask for help.
+
+When the process is consuming 100% CPU, it won't have time to reply to the Readiness probe checks, and it will be eventually removed from the Service.
+
+However, the Pod is actively counted against the active replicas for the current Deployment.
+
+In other words, not only the process isn't serving requests, but also it consumes resources.
+
+You should expose and use a Liveness probe for the current container that replies with a success response.
```
Comment (salmaniqbal) [inline application-development.md]: Not sure if I understand this. Should I use livness probe at all?

File: application-development.md (hunk 1 above)
Comment (danielepolencic) [reply application-development.md]: Readiness is actually what I meant. You have ghost pod because it's running, but not part of the Service. So it consumes one of the slots in the deployment without contributing.

I think I should amend the description and make it more clear.

**Comment by danielepolencic** [comment]
Amended, ready for review
